packages/riipen-ui/src/components/Input.test.jsx

```diff
@@ -0,0 +1,323 @@
+    });
+
+    it("renders hint", () => {
+      const hint = "Test";
+
+      const wrapper = mount(<Input hint={hint} />);
+
+      expect(wrapper.text()).toEqual(hint);
+    });
+
+    it("does not render InputLabel element when hint is not given", () => {
+      const wrapper = mount(<Input />);
+
+      expect(
+        wrapper
+          .find("Input")
+          .children()
+          .contains("InputLabel")
+      ).toEqual(false);
+    });
+  });
+
+  describe("label prop", () => {
+    it("sets valid custom label", () => {
+      const label = "Test";
+
+      const wrapper = mount(<Input label={label} />);
+
+      expect(
+        wrapper
+          .find("Input")
+          .childAt(0)
+          .childAt(0)
+          .type()
+      ).toEqual(InputLabel);
+    });
+
+    it("renders label", () => {
+      const label = "Test";
+
+      const wrapper = mount(<Input label={label} />);
+
+      expect(wrapper.text()).toEqual(label);
+    });
+
+    it("does not render InputLabel element when label is not given", () => {
+      const wrapper = mount(<Input />);
+
+      expect(
+        wrapper
+          .find("Input")
+          .children()
+          .contains("InputLabel")
+      ).toEqual(false);
+    });
+  });
+
+  describe("labelColor prop", () => {
+    it("sets valid custom labelColor", () => {
+      const label = "Test";
+      const labelColor = "black";
+
+      const wrapper = mount(<Input label={label} labelColor={labelColor} />);
+
+      expect(wrapper.find("InputLabel").props().color).toEqual(labelColor);
+    });
+  });
+
+  describe("labelWeight prop", () => {
+    it("sets valid custom labelWeight", () => {
+      const label = "Test";
+      const labelWeight = "bold";
+
+      const wrapper = mount(<Input label={label} labelWeight={labelWeight} />);
+
+      expect(wrapper.find("InputLabel").props().weight).toEqual(labelWeight);
+    });
+  });
+
+  describe("multiline prop", () => {
+    it("sets component as textarea with true multiline prop", () => {
+      const multiline = true;
+
+      const wrapper = mount(<Input multiline={multiline} />);
+
+      expect(
+        wrapper
+          .find("Input")
+          .childAt(0)
+          .childAt(0)
+          .name()
+      ).toEqual("textarea");
+    });
+
+    it("sets component as input with default multiline prop", () => {
+      const wrapper = mount(<Input />);
+
+      expect(
+        wrapper
+          .find("Input")
+          .childAt(0)
+          .childAt(0)
+          .name()
+      ).toEqual("input");
+    });
+  });
+
+  describe("required prop", () => {
+    it("sets valid custom required", () => {
+      const label = "Test";
+      const required = true;
+
+      const wrapper = mount(<Input label={label} required={required} />);
+
+      expect(wrapper.find("InputLabel").props().required).toEqual(required);
+    });
+
+    it("displays an asterisk at the end of the label when required is true", () => {
```

Comment:
I think this test might be actually testing the InputLabel component. The Input component is responsible for passing it into InputLabel and InputLabel does the asterisk.

Reply:
Ya @licarm is right, so I think we can actually remove this test.